candle-core/src/safetensors.rs

```diff
@@ -1,5 +1,61 @@
-use crate::{Device, Error, Result, Tensor, WithDType};
+use crate::{DType, Device, Error, Result, Tensor, WithDType};
 use safetensors::tensor as st;
+use std::borrow::Cow;
+
+impl From<DType> for st::Dtype {
+    fn from(value: DType) -> Self {
+        match value {
+            DType::U8 => st::Dtype::U8,
+            DType::U32 => st::Dtype::U32,
+            DType::BF16 => st::Dtype::BF16,
+            DType::F16 => st::Dtype::F16,
+            DType::F32 => st::Dtype::F32,
+            DType::F64 => st::Dtype::F64,
+        }
+    }
+}
+
+impl TryFrom<st::Dtype> for DType {
+    type Error = Error;
+    fn try_from(value: st::Dtype) -> Result<Self> {
+        match value {
+            st::Dtype::U8 => Ok(DType::U8),
+            st::Dtype::U32 => Ok(DType::U32),
+            st::Dtype::BF16 => Ok(DType::BF16),
+            st::Dtype::F16 => Ok(DType::F16),
+            st::Dtype::F32 => Ok(DType::F32),
+            st::Dtype::F64 => Ok(DType::F64),
+            dtype => Err(Error::UnsupportedSafeTensorDtype(dtype)),
+        }
+    }
+}
+
+impl st::View for Tensor {
+    fn dtype(&self) -> st::Dtype {
+        self.dtype().into()
+    }
+    fn shape(&self) -> &[usize] {
+        self.shape().dims()
+    }
+
+    fn data(&self) -> Cow<[u8]> {
+        // This copies data from GPU to CPU.
+        convert_back(self).unwrap()
+    }
+
+    fn data_len(&self) -> usize {
+        let n: usize = self.shape().elem_count();
+        let bytes_per_element = self.dtype().size_in_bytes();
+        n * bytes_per_element
+    }
+}
+
+impl Tensor {
+    pub fn save<P: AsRef<std::path::Path>>(&self, name: &str, filename: P) -> Result<()> {
```

Comment:
Maybe mention the `safetensors` format in the function name so that it's obvious what gets used?
Also could you expose in the `Tensor` impl some save multi function (similar to what you used in your test below)?

Reply:
`Tensor::save_multi(map, ...)` ? I don't really like that (it's a method on some struct where `serialize_to_file(map,...)` already works.

the `save` is really a convenience method that I'm happy to remove.